spec/grape/middleware/versioner/header_spec.rb

```diff
@@ -184,24 +184,6 @@
       end
     end
 
-    it 'fails with 406 Not Acceptable if type is a range' do
-      expect { subject.call('HTTP_ACCEPT' => '*/*').last }.to raise_exception do |exception|
-        expect(exception).to be_a(Grape::Exceptions::InvalidAcceptHeader)
-        expect(exception.headers).to eql('X-Cascade' => 'pass')
-        expect(exception.status).to eql 406
-        expect(exception.message).to include('Accept header must not contain ranges ("*").')
-      end
-    end
-
-    it 'fails with 406 Not Acceptable if subtype is a range' do
-      expect { subject.call('HTTP_ACCEPT' => 'application/*').last }.to raise_exception do |exception|
-        expect(exception).to be_a(Grape::Exceptions::InvalidAcceptHeader)
-        expect(exception.headers).to eql('X-Cascade' => 'pass')
-        expect(exception.status).to eql 406
-        expect(exception.message).to include('Accept header must not contain ranges ("*").')
-      end
-    end
-
```

Comment:
Yeah, sorry, I tried to explain this in the third paragraph above, but this may need a bit more explanation.

The existing code is doing some checking to make sure that you're not passing in media type ranges.  However, these checks are only happening in the case when `strict` is set to true.  If `strict` is set to true and you're passing in an accept header like `Accept:application/*`, it's my contention that returning an error about not allowing media ranges is less important that returning an error about not passing in a correct version/vendor header.  

As I was refactoring the code, I placed the version/vender header checking mechanism before the media range checks.  After I did this, it became clear that these media range checks were never getting exercised because the version vendor checks were always returning the `406` error first.  So, pulling out the media range checks and the related specs seemed prudent.

Of course, there may be some untested/un-documented outcome that I'm not seeing here related to media type accept headers.  Beyond the README and the specs, is there some kind of written spec or standard that you're intending to follow that might provide more guidance here?